### PR TITLE
extensibility: hide site admin toggle for global settings file users

### LINKED ISSUE #21573: Enabling extensions for all causes a config error
When enabling an extension for all users on sg.com we get an error "Updating global_settings_file". There shouldn't be such an error.

### PR DESCRIPTION
**Pull request by tjkandala**
Fix #21573. We added a toggle for site admins to enable/disable and extension for all users, but settings set through a global settings file cannot be mutated.

- Add `allowSiteSettingsEdits` field to `Site` settings subject type (GraphQL API).
- Hide "enable/disable extension for all" toggle for site admins who use a global settings file.

#### Questions

- When we add new fields to the GraphQL API, how do we handle the case where users are on the latest browser extension version, but point it towards a Sourcegraph instance running an older version? I'm not using this field in the browser extension for now, or probably ever since we don't mutate settings from it, but I wonder how we prevent bugs while iterating on the API (cc @felixfbecker).

### CODE REVIEW

File: cmd/frontend/graphqlbackend/site.go [hunk 1]
```diff
@@ -119,6 +119,12 @@ func (r *siteResolver) ProductSubscription() *productSubscriptionStatus {
 	return &productSubscriptionStatus{}
 }
 
+var allowSiteSettingsEdits = os.Getenv("SITE_CONFIG_FILE") == "" || siteConfigAllowEdits
+
+func (r *siteResolver) AllowSiteSettingsEdits() bool {
+	return allowSiteSettingsEdits
```
Comment (eseliger) [inline cmd/frontend/graphqlbackend/site.go:125]:
```suggestion
	return canUpdateSiteConfiguration()
```

File: client/web/src/extensions/ExtensionsList.tsx [hunk 2]
```diff
@@ -62,10 +62,22 @@ export const ExtensionsList: React.FunctionComponent<Props> = ({
...
-        () => settingsCascade.subjects?.find(settingsSubject => settingsSubject.subject.__typename === 'Site'),
-        [settingsCascade]
-    )
+    const siteSubject = useMemo(() => {
+        if (!settingsCascade.subjects) {
+            return undefined
+        }
+        for (const subject of settingsCascade.subjects) {
+            if (subject.subject.__typename === 'Site') {
+                // Even if the user has permission to admister Site settings, changes cannot be made
```
Comment (eseliger) [inline client/web/src/extensions/ExtensionsList.tsx:71]:
```suggestion
                // Even if the user has permission to administer Site settings, changes cannot be made
```

**Review by eseliger — COMMENTED** [review]
Nice work 🌟

Any chance we can get this integrated into the global settings editor, too? Maybe just as simple as an alert on top of it 🤔

**Comment by tjkandala** [comment]
Thanks @eseliger! I've added `canUpdateSiteConfiguration`

> Any chance we can get this integrated into the global settings editor, too? Maybe just as simple as an alert on top of it

Good idea, I'll tackle this in a separate PR

**Review by eseliger — APPROVED** [review]
🌟

**Pull request merged**